vm/src/vm/vm_new.rs

```diff
@@ -312,4 +314,19 @@ impl VirtualMachine {
             obj.as_object(),
         )
     }
+
+    fn set_exception_attr(
+        &self,
+        exc: PyBaseExceptionRef,
+        attr_name: &str,
+        attr_value: Option<PyObjectRef>,
+    ) -> PyBaseExceptionRef {
+        match exc
+            .as_object()
+            .set_attr(attr_name, self.unwrap_or_none(attr_value), self)
+        {
+            Ok(_) => exc,
+            Err(e) => e,
+        }
+    }
```

Comment:
This function is looking useful in `exceptions.rs`, but I didn't intended this way.